### DISCUSSION TITLE
MongoDB

### DISCUSSION THREAD

**Discussion by lousferig89** [dbeaver/dbeaver#11148] [General]
Hi all! I saw MongoDB listed on the DBeaver overview of supported databases - however it's not among the options I have in the version I have installed (5.3.4).

Do I need to configure something else in terms of drivers etc.?
 [https://9apps.ooo/](https://9apps.ooo/)

**Comment by LonwoLonwo** [comment]
Hello, @lousferig89 
We support almost all NoSQL databases only in the Enterprise version. And MongoDB is on NoSQL list.
You can see information about this on the download page, for example - https://dbeaver.io/download/

**Reply by jankowskico** [reply]
What is stopping someone to implement this feature in CE?

**Reply by p91paul** [reply]
I suppose the developers are not going to merge code that brings a feature like this from EE to CE, as I suspect that support for nosql is a major selling point!

**Reply by tati-kru** [reply]
> What is stopping someone to implement this feature in CE?

The main reason is that DBeaver EE doesn't use JDBC drivers to connect to NoSQL databases. Some NoSQL databases doesn't have JDBC driver at all. But even if they do, their JDBC drivers have a lot of limitations, don't support most of NoSQL features and don't work properly. 

DBeaver EE has own custom connectors for NoSQL databases that help to work with such databases natively. This features are not going to be migrated to DBeaver CE near future. 

However, you always can create a custom connection with your JDBC driver using generic JDBC driver settings as it is described here: https://github.com/dbeaver/dbeaver/wiki/Database-drivers.

**Comment by jankowskico** [comment]
I've managed to take drivers from dbschema.com and add them to dbeaver. Connection works very well though I do not tested it extensively yet.

**Reply by samtech09** [reply]
> I've managed to take drivers from dbschema.com and add them to dbeaver. Connection works very well though I do not tested it extensively yet.

Did you get that working ?

Dbschema JDBC driver is connected, showing databases and tables but not the data.

**Reply by jankowskico** [reply]
It was year ago but I think I got the same results as You. Gave up in the end.

**Reply by crispycrispycrispy** [reply]
Yes this issue with the driver from dbschema.com is still prevalent. :(
